sos/cleaner/parsers/hostname_parser.py

```diff
@@ -82,34 +82,55 @@ def load_hostname_from_etc_hosts(self, content):
                 else:
                     self.mapping.add(host)
 
+    def generate_item_regexes(self):
```

Comment:
> This would work for the hostnames already in a map, and those discovered during `preload_all_archives_into_maps()`, but would not provide coverage for new hosts/domains discovered during an active obfuscation run.

So:
- `generate_item_regexes` should call 2 new methods `generate_shortname_regex` and `generate_domains_regex` that execute relevant parts of this method - these methods will call the `re.compile` and do the sorting
- `generate_shortname_regex` should be called at https://github.com/sosreport/sos/blob/main/sos/cleaner/mappings/hostname_map.py#L220 for the newly added shortname
- `generate_domain_regex` should be called at https://github.com/sosreport/sos/blob/main/sos/cleaner/mappings/hostname_map.py#L243 for the newly added domain
- remove the LGTM warning (orphaned code)

> In addition to this, I think what we'll need to do is add a step to hostname's `add()` method that generates a `re.compile()` object and adds it to this/these lists. 

I dont understand what hostname's `add()` items are not covered?

> On that note, we should probably generate a regex for _both_ the original item (e.g. `example.subdomain.domain.com`) and an underscore-formatted version (`example_subdomain_domain_com`).

Isn't this resolved by https://github.com/sosreport/sos/pull/2844/files#diff-baba7a4d56ffd04533e1ad61d3bb460e35915d843a2cf8cbadcdbafd19f6811bR102 already?

Reply:
> * `generate_shortname_regex` should be called at https://github.com/sosreport/sos/blob/main/sos/cleaner/mappings/hostname_map.py#L220 for the newly added shortname

Ack, or see below.
> * `generate_domain_regex` should be called at https://github.com/sosreport/sos/blob/main/sos/cleaner/mappings/hostname_map.py#L243 for the newly added domain

Ack, or see below.

> I dont understand what hostname's `add()` items are not covered?

What I was saying was that after an item is added to `self.dataset` from the `add()` method, we should from there create the `re.compile()` object and add it to the regex list.

So - we can either do it in place with the `sanitize_item()` flow(s), or within `add()` directly.
> > On that note, we should probably generate a regex for _both_ the original item (e.g. `example.subdomain.domain.com`) and an underscore-formatted version (`example_subdomain_domain_com`).
> 
> Isn't this resolved by https://github.com/sosreport/sos/pull/2844/files#diff-baba7a4d56ffd04533e1ad61d3bb460e35915d843a2cf8cbadcdbafd19f6811bR102 already?

In the current implementation it only does so for the `_` re-formatted strings, relying on the default behavior for dotted style. With a change like this, we'll need an `re.compile()` object for both `foo.example.com` and `foo_example_com` to ensure we hit both formatting styles.